src/output/dnssim.lua

```diff
@@ -125,13 +125,59 @@ end
 --
 -- Refer to:
 -- .I https://gnutls.org/manual/html_node/Priority-Strings.html
-function DnsSim:tls(priority)
-    if priority ~= nil then
-        C.output_dnssim_tls_priority(self.obj, priority)
+function DnsSim:tls(tls_priority)
+    if tls_priority ~= nil then
+        C.output_dnssim_tls_priority(self.obj, tls_priority)
     end
     C.output_dnssim_set_transport(self.obj, C.OUTPUT_DNSSIM_TRANSPORT_TLS)
 end
 
+-- Set the transport to HTTP/2 over TLS.
+-- The following
+-- .B http2_options
+-- can be set:
+--
+-- .B method:
+-- .B GET
+-- or
+-- .B POST
+-- (default)
+--
+-- .B uri_path:
+-- where queries will be sent.
+-- Defaults to
+-- .B /dns-query
+--
+-- .B zero_out_msgig:
+-- when
+-- .B true
+-- (default), query ID is always set to 0
+function DnsSim:https2(http2_options, tls_priority)
+    if tls_priority ~= nil then
+        C.output_dnssim_tls_priority(self.obj, tls_priority)
+    end
+
+    uri_path = "/dns-query"
+    zero_out_msgid = true
+    method = "GET"
+    if http2_options ~= nil and type(http2_options) == "table" then
```

Comment:
This will silently pass if `http2_options` is not a table.

Might be better to check if it's a table (if not nil) and go fatal if not.

Also document that it's a table, and add an example.

Reply:
Done. Overall I'm not sure whether fatal or critical log level is better in cases like this. I decided to go with critical to be consistent with the other lua modules.

Reply:
fatal!

Where is this critical in other modules?

Reply:
Seems like a problem I've created myself, I've adjusted the log levels from critical -> fatal for invalid settings.